### DISCUSSION TITLE
tri-layer / conditional layers support

### DISCUSSION THREAD

**Discussion by luked42** [nickcoutsos/keymap-editor#64] [Q&A] [answered]
hi, 

is there tri-layer support in the tool? I wasn't able to find the functionality
https://zmk.dev/docs/features/conditional-layers

best,
Luke

**Comment by nickcoutsos** [comment, marked as answer]
**Update: this feature exists, see the below post**

Hi there,

No support currently, I actually haven't looked at this feature before. It seems fairly straightforward though I don't have immediate plans to implement an editor for it.

I say this a lot but, for what it's worth, if conditional layers are something you'd want to use but not edit frequently you can still use keymap editor to make other changes while keeping other devicetree nodes intact.

**Comment by nickcoutsos** [comment]
I forget to come back to feature requests in the Discussions tab soemtimes -- if you haven't seen it already I've added support for conditional layers a little while back! This includes automatically updating a conditional layer node when a referenced layer gets reordered or deleted.

One limitation is that when the `then-layer` gets deleted (or, I guess, all of the `if-layers`) then you get into a state where the keymap won't build. I was considering just deleting the condition in that case but it feels wrong to do that without prompting the user if they feel it would be easier to go manually correct it than recreate.
